### PR TITLE
Fix/timeseries generation

### PR DESCRIPTION
**Pull request by pennfranc**
Implemented changes requested by Julien, including:
- changed the default value of constant_timeseries 
- renamed 'offset' to 'freq' and 'start_date' to 'start_ts' 
- improvements to docstrings 
- replaced parameter 'value_delta' with 'end_value' in linear_timeseries 
- made random_walk_timeseries more efficient 
- renamed white_noise_timeseries to gaussian_timeseries 
- added multivariate gaussian noise support
- renamed periodic_timeseries to sine_timeseries

### CODE REVIEW

File: u8timeseries/utils/timeseries_generation.py [hunk 1]
```diff
@@ -4,96 +4,111 @@
...
+    times = pd.date_range(periods=length, freq=freq, start=start_ts)
+    values = np.linspace(start_value, end_value, length)
 
     return TimeSeries.from_times_and_values(times, values)
 
 
-def periodic_timeseries(frequency: float = 0.1, amplitude: float = 1, phase: float = 0, y_offset: float = 0,
-                        length: int = 10, offset: str = 'D',
-                        start_date: pd.Timestamp = pd.Timestamp('2000-01-01')) -> 'TimeSeries':
+def periodic_timeseries(value_frequency: float = 0.1, value_amplitude: float = 1, value_phase: float = 0,
```
Comment (hrzn) [inline u8timeseries/utils/timeseries_generation.py:53]: Here I think we could even name it `sine_timeseries`as not all periodic time series are sinusoids.

File: u8timeseries/utils/timeseries_generation.py [hunk 2]
```diff
@@ -4,96 +4,111 @@
...
+                        std: Union[float, np.ndarray] = 1, start_ts: pd.Timestamp = pd.Timestamp('2000-01-01')) -> 'TimeSeries':
     """
-    Creates a white noise TimeSeries by sampling a gaussian distribution with mean 'mean' and 
-    standard deviation 'std'. Each value represents an indipendent sample of the distribution.
+    Creates a gaussian noise TimeSeries by sampling a gaussian distribution with mean 'mean' and 
+    standard deviation 'std'. Each value represents a sample of the distribution.
+    When the mean is set to 0, it can be considered a white noise TimeSeries.
 
     :param mean: The mean of the gaussian distribution that is sampled at each step.
+                 If a float value is given, the same mean is used at every step.
```
Comment (hrzn) [inline u8timeseries/utils/timeseries_generation.py:85]: cool

File: u8timeseries/utils/timeseries_generation.py [hunk 3]
```diff
@@ -102,30 +117,28 @@ def random_walk_timeseries(length: int = 10, offset: str = 'D', mean: float = 0,
...
+    :param start_ts: The time index of the first entry in the returned TimeSeries.
     :return: A random walk TimeSeries created as indicated above.
     """
 
-    times = pd.date_range(periods=length, freq=offset, start=start_date)
-    values = [np.random.normal(mean, std)]
-    while (len(values) < length):
-        values.append(values[-1] + np.random.normal(mean, std))
+    times = pd.date_range(periods=length, freq=freq, start=start_ts)
+    values = np.cumsum(np.random.normal(mean, std, size=length))
```
Comment (hrzn) [inline u8timeseries/utils/timeseries_generation.py:126]: nice :)

File: u8timeseries/utils/timeseries_generation.py (hunk 1 above)
Comment (pennfranc) [reply u8timeseries/utils/timeseries_generation.py]: Done

**Pull request merged**